### PR TITLE
feat(port): Bionics can require another bionic

### PR DESCRIPTION
**Pull request by Zlorthishen**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change
ports 
 - https://github.com/CleverRaven/Cataclysm-DDA/pull/66996

Fixes https://github.com/cataclysmbnteam/Cataclysm-BN/issues/5455

## Describe the solution
ports the dda change which can allow bionic  A to require bionic B to already be installed prior to installation of Bionic A

## Testing
temp1
temp2
temp3
image

### CODE REVIEW

File: doc/src/content/docs/en/mod/json/reference/json_info.md [hunk 1]
```diff
@@ -188,43 +188,56 @@ the appropriate JSON file.
...
+| act_cost        | (_optional_) How many kJ it costs to activate the bionic. Strings can be used "1 kJ"/"1000 J"/"1000000 mJ" (default: `0`)                                                       |
+| deact_cost      | (_optional_) How many kJ it costs to deactivate the bionic. Strings can be used "1 kJ"/"1000 J"/"1000000 mJ" (default: `0`)                                                     |
+| react_cost      | (_optional_) How many kJ it costs over time to keep this bionic active, does nothing without a non-zero "time". Strings can be used "1 kJ"/"1000 J"/"1000000 mJ" (default: `0`) |
+| time            | (_optional_) How long, when activated, between drawing cost. If 0, it draws power once. (default: `0`)                                                                          |
+| upgraded_bionic | (_optional_) Bionic that can be upgraded by installing this one.                                                                                                                |
+| required_bionic | (_optional_) Bionic which is required to install this bionic, and which cannot be uninstalled if this bionic is installed                                                       |
+
```
Comment (scarf005) [inline doc/src/content/docs/en/mod/json/reference/json_info.md:203]:
```suggestion
```

could this line be removed? it's breaking markdown table formatting.

**Review by scarf005 — DISMISSED** [review]

**Review by chaosvolt — CHANGES_REQUESTED** [review]
I might first advise checking for professions that use hydraulic muscles and add the needed CBM to them, just to be safe.

**Comment by scarf005** [comment]
> I might first advise checking for professions that use hydraulic muscles and add the needed CBM to them, just to be safe.

or even better, maybe auto-add required bionics in following PR?

**Review by chaosvolt — APPROVED** [review]
Aight, looks good now, compiles and starting as a cyborg with the relevant CBMs doesn't do anything weird.

**Pull request merged**